test/evp_test.c

```diff
@@ -2531,17 +2524,11 @@ static int parse(EVP_TEST *t)
             return 0;
         key->name = take_value(pp);
 
-        /* Hack to detect SM2 keys */
-        if(strstr(key->name, "SM2"))
-            EVP_PKEY_set_alias_type(pkey, EVP_PKEY_SM2);
-
         key->key = pkey;
```

Comment:
When pkey = NULL, tests fail because the key can be found, and it's NULL, so operations fail. At least, that's what I saw. I will download this to see if it fixes the problem I saw.